iroh/src/commands/doctor.rs

```diff
@@ -600,6 +619,124 @@ async fn port_map(local_port: NonZeroU16, timeout: Duration) -> anyhow::Result<(
     }
 }
 
+async fn derp_regions(config: Config) -> anyhow::Result<()> {
+    let key = iroh_net::hp::key::node::SecretKey::generate();
+    let mut set = tokio::task::JoinSet::new();
+    if config.derp_regions.is_empty() {
+        println!("No DERP Regions specified in the config file.");
+    }
+    for region in config.derp_regions.into_iter() {
+        let secret_key = key.clone();
+        set.spawn(async move {
+            let mut region_details = RegionDetails {
+                latency: None,
+                error: None,
+                region_id: region.region_id,
+                hosts: region.nodes.iter().map(|n| n.url.clone()).collect(),
+            };
+            let client = match iroh_net::hp::derp::http::ClientBuilder::new()
+                .get_region(move || {
+                    let region = region.clone();
+                    Box::pin(async move { Some(region) })
+                })
+                .build(secret_key)
+            {
+                Ok(c) => c,
+                Err(e) => {
+                    region_details.error = Some(e.to_string());
+                    return region_details;
+                }
+            };
+            let start = std::time::Instant::now();
+            match tokio::time::timeout(Duration::from_secs(2), client.connect()).await {
+                Err(e) => {
+                    region_details.error = Some(e.to_string());
+                }
+                Ok(Err(e)) => {
+                    region_details.error = Some(e.to_string());
+                }
+                Ok(_) => {
+                    region_details.latency = Some(start.elapsed());
+                }
+            }
+            region_details
+        });
+    }
+    let mut success = Vec::new();
+    let mut fail = Vec::new();
+    while let Some(region_details) = set.join_next().await {
+        let region_details = region_details?;
+        if region_details.latency.is_some() {
+            success.push(region_details);
+        } else {
+            fail.push(region_details);
+        }
+    }
+    success.sort_by(|a, b| {
```

Comment:
I think you can use https://doc.rust-lang.org/std/primitive.slice.html#method.sort_by_key here